tests/integration/test_record.py

```diff
@@ -45,85 +45,50 @@ def test_incoming_deleted_record(app):
-               '	</datafield>'
-               '     <datafield tag="701" ind1=" " ind2=" ">'
-               '         <subfield code="a">'
-               '             Baer, Howard'
-               '         </subfield>'
-               '     </datafield>'
-               '     <datafield tag="999" ind1="C" ind2="6">'
-               '         <subfield code="a">0-1-0-1-0-0-1</subfield>'
-               '         <subfield code="t">2016-09-13 20:31:28</subfield>'
-               '         <subfield code="v">Invenio/1.1.2.1260-aa76f refextract/1.5.44</subfield>'
-               '         <subfield code="v">tmpbzPvDR.pdf</subfield>'
-               '     </datafield>'
-               '     <datafield tag="245" ind1=" " ind2=" ">'
-               '         <subfield code="a">Mixed axion-neutralino dark mater in the supersymmetric DFSZ mOD</subfield>'
-               '     </datafield>'
-               '     <datafield tag="024" ind1="7" ind2=" ">'
-               '         <subfield code="2">HDL</subfield>'
-               '         <subfield code="a">11244/44597</subfield>'
-               '     </datafield>'
-               '     <datafield tag="035" ind1=" " ind2=" ">'
-               '         <subfield code="a">Serce:2016ygf</subfield>'
-               '         <subfield code="9">INSPIRETeX</subfield>'
-               '     </datafield>'
-               '     <datafield tag="100" ind1=" " ind2=" ">'
-               '         <subfield code="a">Serce, Hasan Baris</subfield>'
-               '         <subfield code="u">Oklahoma U.</subfield>'
-               '     </datafield>'
-               '</record>')
-
     json_record = clean_record(hepnames.do(create_record(snippet)))
-
     json_record['$schema'] = "http://localhost:5000/schemas/records/hep.json"
 
-    record = record_upsert(json_record)
-
-    r = RecordIndexer()
-    r.index(record)
-
-    db.session.commit()
-
-    ########################
-
+    record = insert_record_json(json_record)
     record['deleted'] = True
 
-    new_record = record_upsert(record)
+    new_record = insert_record_json(record)
 
-    # import pdb
-    # pdb.set_trace()
+    with app.test_client() as client:
+        response = client.get('/api/literature/1486322')
+        assert response.status_code == 410
 
-    r = RecordIndexer()
-    r.index(new_record)
 
-    db.session.commit()
+def insert_record_json(json_record):
+    index_queue = []
+    try:
+        with db.session.begin_nested():
+            record = record_upsert(json_record)
+            if record:
+                index_queue.append(create_index_op(record))
+        db.session.commit()
+    finally:
+        db.session.close()
 
-    with app.test_client() as client:
-        response = client.get('/api/literature/1486323')
+    req_timeout = current_app.config['INDEXER_BULK_REQUEST_TIMEOUT']
+    es_bulk(
+        current_search_client,
+        index_queue,
+        stats_only=True,
+        request_timeout=req_timeout,
+    )
```

Comment:
This whole technique is using `es_bulk` for one record. In this case it might be overkill.
Just use `invenio_indexer.api.RecordIndexer().index(record)`